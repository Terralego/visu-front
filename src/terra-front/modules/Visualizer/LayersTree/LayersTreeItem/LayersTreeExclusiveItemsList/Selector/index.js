import Selector from './Selector';
import { connectLayersTree } from '../../../LayersTreeProvider/context';

export default connectLayersTree(({
  getLayerState, translate,
}, {
  layer: { layers, selectors },
}) => ({
  selectors,
  layers,
  activeLayer: layers.find(layer => getLayerState({ layer }).active),
  translate,
}))(Selector);
