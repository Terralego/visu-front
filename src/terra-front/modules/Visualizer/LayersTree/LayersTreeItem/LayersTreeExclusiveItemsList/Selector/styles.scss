.selector {
  &__message {
    margin-top: .35rem;
  }
}
