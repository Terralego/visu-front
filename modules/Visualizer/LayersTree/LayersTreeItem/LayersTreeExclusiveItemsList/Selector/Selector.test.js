import React from 'react';
import renderer from 'react-test-renderer';

import Selector from './Selector';

it('should render', () => {
  const tree = renderer.create(
    <>
      <Selector
        selectors={[{
          label: 'Foo',
          name: 'foo',
          values: ['foo', 'bar'],
        }, {
          label: 'Bar',
          name: 'bar',
          values: ['foo', 'bar'],
        }]}
        layers={[{
          label: 'Layer 1',
          selectorKey: {
            foo: 'foo',
            bar: 'foo',
          },
        }, {
          label: 'Layer 1',
          selectorKey: {
            foo: 'foo',
            bar: 'bar',
          },
        }]}
        activeLayer={{
          selectorKey: {
            foo: 'foo',
            bar: 'foo',
          },
        }}
      />
    </>,
  );
  expect(tree.toJSON).toMatchSnapshot();
});

it('should render an error message', () => {
  const tree = renderer.create(
    <>
      <Selector
        selectors={[{
          label: 'Foo',
          name: 'foo',
          values: ['foo', 'bar'],
        }, {
          label: 'Bar',
          name: 'bar',
          values: ['foo', 'bar'],
        }]}
        layers={[{
          label: 'Layer 1',
          selectorKey: {
            foo: 'foo',
            bar: 'foo',
          },
        }, {
          label: 'Layer 1',
          selectorKey: {
            foo: 'foo',
            bar: 'bar',
          },
        }]}
        activeLayer={{
          selectorKey: {
            foo: 'foo',
            bar: 'foo',
          },
        }}
      />
    </>,
  );
  tree.getInstance().setState({ noMatchingLayer: true });
  expect(tree.toJSON).toMatchSnapshot();
});

it('should change layer', () => {
  const selectors = [{
    label: 'Foo',
    name: 'foo',
    values: ['foo', 'bar'],
  }, {
    label: 'Bar',
    name: 'bar',
    values: ['foo', 'bar'],
  }];
  const layers = [{
    label: 'Layer 1',
    selectorKey: {
      foo: 'foo',
      bar: 'foo',
    },
  }, {
    label: 'Layer 1',
    selectorKey: {
      foo: 'foo',
      bar: 'bar',
    },
  }];
  const onChange = jest.fn();
  const instance = new Selector({
    selectors,
    layers,
    activeLayer: layers[0],
    onChange,
  });
  instance.setState = jest.fn();
  instance.onChange('bar')('bar');
  expect(onChange).toHaveBeenCalledWith(1);
  expect(instance.setState).toHaveBeenCalledWith({ noMatchingLayer: false });
  onChange.mockClear();
  instance.setState.mockClear();

  instance.onChange('bar')('lorem');
  expect(onChange).not.toHaveBeenCalledWith(1);
  expect(instance.setState).toHaveBeenCalledWith({ noMatchingLayer: true });
});
