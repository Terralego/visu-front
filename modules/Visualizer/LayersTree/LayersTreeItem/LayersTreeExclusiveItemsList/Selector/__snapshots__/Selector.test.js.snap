// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `[Function]`;

exports[`should render an error message 1`] = `[Function]`;
